app/reducers.js

```diff
@@ -51,6 +56,11 @@ export function deleteDestinationReducer(state, action) {
   });
 }
 
+export function addHouseReducer(state, action) {
+  const houses = state.houses.set(uuid.v1(), Map({ url: action.url }));
```

Comment:
I'm not 100% sure but you might not need the `Map` around the last object, ImmutableJS might do that conversion anyway...